### ISSUE TITLE
warning pydantic v2: `pydantic.generics:GenericModel` has been moved to `pydantic.BaseModel`

### ISSUE THREAD

**Issue by bVdCreations** [uriyyo/fastapi-pagination#784]
I recently upgraded to pydantic V2 and there is a small migration warning popping up from this library

```bash
src/main.py:10: in <module>
    from fastapi_pagination import add_pagination
/usr/local/lib/python3.11/site-packages/fastapi_pagination/__init__.py:16: in <module>
    from .api import (
/usr/local/lib/python3.11/site-packages/fastapi_pagination/api.py:41: in <module>
    from .bases import AbstractPage, AbstractParams
/usr/local/lib/python3.11/site-packages/fastapi_pagination/bases.py:44: in <module>
    from pydantic.generics import GenericModel
/usr/local/lib/python3.11/site-packages/pydantic/_migration.py:276: in wrapper
    warnings.warn(f'`{import_path}` has been moved to `{new_location}`.')
E   UserWarning: `pydantic.generics:GenericModel` has been moved to `pydantic.BaseModel`.
```

**Comment by uriyyo** [comment]
Hi @bVdCreations,

This issue should be fixed in version `0.12.8`.

**Comment by uriyyo** [comment]
Hi @bVdCreations,

I am closing this issue, please reopen it in case if the issue will appear again.
